include/z64bgcheck.h

```diff
@@ -0,0 +1,203 @@
+    union {
+        u16 vtxData[3];
+        struct {
+            /* 0x02 */ u16 flags_vIA; // 0xE000 is poly exclusion flags, 0x1FFF is vtxId
+            /* 0x04 */ u16 flags_vIB; // 0xE000 is ? flags, 0x1FFF is vtxId
+            /* 0x06 */ u16 vIC;
+        };
+    };
+    /* 0x08 */ Vec3s norm; // Unit normal vector.
+                           // Value ranges from -0x7FFF to 0x7FFF, representing -1 to 1. 0x8000 is invalid
+
+    /* 0x0E */ s16 dist; // Plane distance from origin
+} CollisionPoly;         // size = 0x10
+
+typedef struct {
+    /* 0x00 */ u16 cameraSType;
+    /* 0x02 */ s16 numCameras;
+    /* 0x04 */ Vec3s* camPosData;
+} CamData;
+
+typedef struct {
+    /* 0x00 */ s16 xMin;
+    /* 0x02 */ s16 ySurface;
+    /* 0x04 */ s16 zMin;
+    /* 0x06 */ s16 xLength;
+    /* 0x08 */ s16 zLength;
+    /* 0x0C */ u32 properties;
+
+    // 0x0008_0000 = ?
+    // 0x0007_E000 = Room Index, 0x3F = all rooms
+    // 0x0000_1F00 = Lighting Settings Index
+    // 0x0000_00FF = CamData index
+} WaterBox; // size = 0x10
+
+typedef struct {
+    u32 data[2];
+} SurfaceType;
+// Type 1
+// 0x0800_0000 = wall damage
+
+typedef struct {
+    /* 0x00 */ Vec3s minBounds; // colAbsMin
+    /* 0x06 */ Vec3s maxBounds; // colAbsMax
+    /* 0x0C */ u16 nbVertices;
+    /* 0x10 */ Vec3s* vtxList; // vertexArray
+    /* 0x14 */ u16 nbPolygons;
+    /* 0x18 */ CollisionPoly* polyList; // polygonArray
+    /* 0x1C */ SurfaceType* polygonTypes;
+    /* 0x20 */ CamData* cameraDataList;
+    /* 0x24 */ u16 nbWaterBoxes;
+    /* 0x28 */ WaterBox* waterBoxes;
+} CollisionHeader; // BGDataInfo
+
+typedef struct {
+    s16 polyId;
+    u16 next; // next SSNode index
+} SSNode;
+
+typedef struct {
+    /* 0x00 */ u16 max;          // original name: short_slist_node_size
+    /* 0x02 */ u16 count;        // original name: short_slist_node_last_index
+    /* 0x04 */ SSNode* tbl;      // original name: short_slist_node_tbl
+    /* 0x08 */ u8* polyCheckTbl; // set to 1 if polygon has already been tested
+} SSNodeList;
+
+typedef struct {
+    SSNode* tbl;
+    s32 count;
+    s32 max;
+} DynaSSNodeList;
+
+typedef struct {
+    u16 floor;
+    u16 wall;
+    u16 ceiling;
+} Lookup;
+
+typedef struct {
+    u16 polyStartIndex;
+    u16 ceiling;
+    u16 wall;
+    u16 floor;
+} DynaLookup;
+
+typedef struct {
+    /* 0x00 */ struct Actor* actor;
+    /* 0x04 */ CollisionHeader* colHeader;
+    /* 0x08 */ DynaLookup dynaLookup;
+    /* 0x10 */ u16 vtxStartIndex; // dyna vtx index start
+    /* 0x14 */ ScaleRotPos prevSrp;
+    /* 0x34 */ ScaleRotPos curSrp;
+    /* 0x54 */ Sphere16 boundingSphere;
+    /* 0x5C */ f32 minY;
+    /* 0x60 */ f32 maxY;
+} BgActor; // size = 0x64
+
+typedef struct {
+    /* 0x0000 */ u8 bitFlag;
+    /* 0x0004 */ BgActor bgActors[BG_ACTOR_MAX];
+    /* 0x138C */ u16 bgActorFlags[BG_ACTOR_MAX]; // & 0x0008 = no dyna ceiling
+    /* 0x13F0 */ CollisionPoly* polyList;        // pbuf
+    /* 0x13F4 */ Vec3s* vtxList;                 // pbuf
```

Comment:
delete i think? if its the original name for something then label which with "original name: " or something like that